torchtune/utils/_perf_profiler.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import ContextManager, Optional
+
+import torch
+from torch.profiler import profile
+
+
+def perf_profiler(
+    output_dir: Optional[str] = "./torchtune_perf_tracing.json",
+) -> ContextManager:
+    """
+    Utility component to trace the code with pytorch profiler.
+    check the user manual: https://pytorch.org/docs/stable/profiler.html for more details.
+    The schedule for this profiler is wait 5 steps, warmup 5 steps, trace 5 steps
+    Note: Enable pytorch profiler may casue performance overhead.
```

Comment:
```suggestion
    Note: Enabling pytorch profiler may add performance overhead.
```

Do you know if this is additional memory or a speed reduction?

Reply:
Mainly speed reduction. Will update the comment to be more specific